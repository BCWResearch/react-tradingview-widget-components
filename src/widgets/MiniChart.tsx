import React, {
	useEffect, useRef 
} from "react";
import { MiniChartProps } from "../types/MiniChartProps";

export const MiniChart = ({ containerStyles, lineColor, topGradientColor, bottomGradientColor, width, height, autosize, ...props }: Partial<MiniChartProps>) => {
	const containerRef = useRef<HTMLDivElement | null>(null);

	useEffect(() => {
		if (!containerRef.current) return;
		const script = document.createElement("script");
		script.type = "text/javascript";
		script.src = "https://s3.tradingview.com/external-embedding/embed-widget-mini-symbol-overview.js";
		script.async = true;
		script.onload = async () => {
			const iframe = containerRef.current?.querySelector("iframe");
			if (iframe && iframe instanceof Element) iframe.style.colorScheme = "normal";
		};
		const config = JSON.stringify({
			width: autosize ? "100%" : (width || 350),
			height: autosize ? "100%" : (height || 220),
			dateRange: "1D",
			colorTheme: "dark",
			trendLineColor: lineColor || "rgba(41, 98, 255, 1)",
			underLineColor: topGradientColor || "rgba(41, 98, 255, 0.3)",
			underLineBottomColor: bottomGradientColor || "rgba(41, 98, 255, 0)",
			isTransparent: true,
			...props,
		});
		script.innerHTML = config;
		containerRef.current.appendChild(script);
		return () => {
			while (containerRef.current?.firstChild) {
				containerRef.current.removeChild(containerRef.current.firstChild);
			}
		};
	}, [ JSON.stringify(props) ]);

	return (
		<div style={containerStyles} className="tradingview-widget-container" ref={containerRef}/>
	);
};