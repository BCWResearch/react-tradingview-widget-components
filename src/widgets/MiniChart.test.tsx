import { MiniChart } from "./MiniChart";
import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

test("should render without crashing", () => {
	const { container } = render(<MiniChart/>);
	expect(container).toMatchSnapshot();
});