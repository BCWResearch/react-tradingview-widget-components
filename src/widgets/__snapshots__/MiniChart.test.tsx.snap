// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div>
  <div
    class="tradingview-widget-container"
  >
    <script
      src="https://s3.tradingview.com/external-embedding/embed-widget-mini-symbol-overview.js"
      type="text/javascript"
    >
      {"width":350,"height":220,"dateRange":"1D","colorTheme":"dark","trendLineColor":"rgba(41, 98, 255, 1)","underLineColor":"rgba(41, 98, 255, 0.3)","underLineBottomColor":"rgba(41, 98, 255, 0)","isTransparent":true}
    </script>
  </div>
</div>
`;
